src/odemis/acq/stream/_helper.py

```diff
@@ -219,47 +256,36 @@ def _updateROIAndPixelSize(self, roi, pxs):
             _, rep, pxs = self._updateROIAndPixelSize((0, 0, 1, 1), pxs)
             return roi, rep, pxs
 
-        # TODO: use the fact that pxs_range/fov is fixed => faster
-        pxs_range = self._getPixelSizeRange()
-        pxs = max(pxs_range[0], min(pxs, pxs_range[1]))
-
         roi = self._fitROI(roi)
 
-        # compute the repetition (ints) that fits the ROI with the pixel size
+        # Compute scale based on dim X, and ensure it's within range
         epxs = self._scanner.pixelSize.value
+        scale = pxs / epxs[0]
+        min_scale = max(self._scanner.scale.range[0])
+        max_scale = min(self._scanner.shape)
+        scale = max(min_scale, min(scale, max_scale))
+        pxs = scale * epxs[0]
+
+        # compute the repetition (ints) that fits the ROI with the pixel size
         eshape = self._scanner.shape
```

Comment:
Originally, the "scanner" was the "emitter" (and it's still often the same hardware in practice). The emitter used to always be the e-beam. So that "e" stood for e-beam or emitter. Of course, now that there is also the confocal hardware, named "laser-mirror", it's less obvious. I'll rename epxs and eshape to spxs and sshape.